Convert int to size_t in for-loops

Here are many warning messages such as:
```
INFO: From Compiling tensorflow_quantum/core/src/program_resolution.cc:
tensorflow_quantum/core/src/program_resolution.cc: In function 'tensorflow::Status tfq::ResolveQubitIds(cirq::google::api::v2::Program*, std::vector<tfq::proto::PauliSum>*)':
tensorflow_quantum/core/src/program_resolution.cc:61:21: warning: comparison of integer expressions of different signedness: 'int' and 'std::vector<std::basic_string<char> >::size_type' {aka 'long unsigned int'} [-Wsign-compare]
   61 |   for (int i = 0; i < ids.size(); i++) {
      |                   ~~^~~~~~~~~~~~
tensorflow_quantum/core/src/program_resolution.cc:77:23: warning: comparison of integer expressions of different signedness: 'int' and 'std::vector<tfq::proto::PauliSum>::size_type' {aka 'long unsigned int'} [-Wsign-compare]
   77 |     for (int i = 0; i < p_sums->size(); i++) {
      |                     ~~^~~~~~~~~~~~~~~~
INFO: From Compiling tensorflow_quantum/core/ops/parse_context.cc:
tensorflow_quantum/core/ops/parse_context.cc: In function 'tensorflow::Status tfq::GetProgramsAndNumQubits(tensorflow::OpKernelContext*, std::vector<cirq::google::api::v2::Program>*, std::vector<int>*, std::vector<std::vector<tfq::proto::PauliSum> >*)':
tensorflow_quantum/core/ops/parse_context.cc:132:21: warning: comparison of integer expressions of different signedness: 'int' and 'std::vector<cirq::google::api::v2::Program>::size_type' {aka 'long unsigned int'} [-Wsign-compare]
  132 |   for (int i = 0; i < programs->size(); i++) {
      |                   ~~^~~~~~~~~~~~~~~~~~
INFO: From Compiling tensorflow_quantum/core/ops/tfq_ps_weights_from_symbols_op.cc:
tensorflow_quantum/core/ops/tfq_ps_weights_from_symbols_op.cc: In member function 'virtual void tfq::TfqPsWeightsFromSymbolOp::Compute(tensorflow::OpKernelContext*)':
tensorflow_quantum/core/ops/tfq_ps_weights_from_symbols_op.cc:127:23: warning: comparison of integer expressions of different signedness: 'int' and 'std::vector<int>::size_type' {aka 'long unsigned int'} [-Wsign-compare]
  127 |     for (int i = 0; i < n_single_symbol.size(); i++) {
      |                     ~~^~~~~~~~~~~~~~~~~~~~~~~~
tensorflow_quantum/core/ops/tfq_ps_weights_from_symbols_op.cc: In lambda function:
tensorflow_quantum/core/ops/tfq_ps_weights_from_symbols_op.cc:147:29: warning: comparison of integer expressions of different signedness: 'int' and 'std::vector<float>::size_type' {aka 'long unsigned int'} [-Wsign-compare]
  147 |           for (int k = 0; k < output_results.at(i).at(j).size(); k++) {
      |                           ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
INFO: From Compiling tensorflow_quantum/core/ops/tfq_ps_symbol_replace_op.cc:
tensorflow_quantum/core/ops/tfq_ps_symbol_replace_op.cc: In member function 'virtual void tfq::TfqPsSymbolReplaceOp::Compute(tensorflow::OpKernelContext*)':
tensorflow_quantum/core/ops/tfq_ps_symbol_replace_op.cc:136:28: warning: comparison of integer expressions of different signedness: 'size_t' {aka 'long unsigned int'} and 'const int' [-Wsign-compare]
  136 |       for (size_t j = 0; j < n_symbols; j++) {
      |                          ~~^~~~~~~~~~~
tensorflow_quantum/core/ops/tfq_ps_symbol_replace_op.cc: In lambda function:
tensorflow_quantum/core/ops/tfq_ps_symbol_replace_op.cc:160:27: warning: comparison of integer expressions of different signedness: 'int' and 'std::vector<std::basic_string<char> >::size_type' {aka 'long unsigned int'} [-Wsign-compare]
  160 |         for (int j = 0; j < output_programs.at(pidx).at(sidx).size(); j++) {
      |                         ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tensorflow_quantum/core/ops/tfq_ps_symbol_replace_op.cc:164:66: warning: comparison of integer expressions of different signedness: 'int' and 'size_t' {aka 'long unsigned int'} [-Wsign-compare]
  164 |         for (int j = output_programs.at(pidx).at(sidx).size(); j < biggest_pad;
      |                                                                ~~^~~~~~~~~~~~~
INFO: From Compiling tensorflow_quantum/core/src/program_resolution.cc:
tensorflow_quantum/core/src/program_resolution.cc: In function 'tensorflow::Status tfq::ResolveQubitIds(cirq::google::api::v2::Program*, std::vector<tfq::proto::PauliSum>*)':
tensorflow_quantum/core/src/program_resolution.cc:61:21: warning: comparison of integer expressions of different signedness: 'int' and 'std::vector<std::basic_string<char> >::size_type' {aka 'long unsigned int'} [-Wsign-compare]
   61 |   for (int i = 0; i < ids.size(); i++) {
      |                   ~~^~~~~~~~~~~~
tensorflow_quantum/core/src/program_resolution.cc:77:23: warning: comparison of integer expressions of different signedness: 'int' and 'std::vector<tfq::proto::PauliSum>::size_type' {aka 'long unsigned int'} [-Wsign-compare]
   77 |     for (int i = 0; i < p_sums->size(); i++) {
      |                     ~~^~~~~~~~~~~~~~~~
```
Let's use `size_t` instead of `int` in C++.

--- a/tensorflow_quantum/core/ops/parse_context.cc
+++ b/tensorflow_quantum/core/ops/parse_context.cc
@@ -129,7 +129,7 @@ Status GetProgramsAndNumQubits(
   }
 
   num_qubits->reserve(programs->size());
-  for (int i = 0; i < programs->size(); i++) {
+  for (size_t i = 0; i < programs->size(); i++) {
     Program& program = (*programs)[i];
     Status status = Status::OK();
     if (p_sums) {

--- a/tensorflow_quantum/core/ops/tfq_ps_symbol_replace_op.cc
+++ b/tensorflow_quantum/core/ops/tfq_ps_symbol_replace_op.cc
@@ -58,7 +58,7 @@ class TfqPsSymbolReplaceOp : public tensorflow::OpKernel {
             "symbols must be rank 1. Got rank ", symbols_tensor->dims(), ".")));
 
     const auto symbols = symbols_tensor->vec<std::string>();
-    const int n_symbols = symbols.size();
+    const size_t n_symbols = symbols.size();
 
     // Parse the replacement string here.
     const Tensor *replacement_symbols_tensor;
@@ -157,12 +157,12 @@ class TfqPsSymbolReplaceOp : public tensorflow::OpKernel {
       for (int i = start; i < end; i++) {
         int sidx = i % n_symbols;
         int pidx = i / n_symbols;
-        for (int j = 0; j < output_programs.at(pidx).at(sidx).size(); j++) {
+        for (size_t j = 0; j < output_programs.at(pidx).at(sidx).size(); j++) {
           output_tensor(pidx, sidx, j) =
               output_programs.at(pidx).at(sidx).at(j);
         }
-        for (int j = output_programs.at(pidx).at(sidx).size(); j < biggest_pad;
-             j++) {
+        for (size_t j = output_programs.at(pidx).at(sidx).size();
+             j < biggest_pad; j++) {
           output_tensor(pidx, sidx, j) = empty_program;
         }
       }

--- a/tensorflow_quantum/core/ops/tfq_ps_weights_from_symbols_op.cc
+++ b/tensorflow_quantum/core/ops/tfq_ps_weights_from_symbols_op.cc
@@ -124,7 +124,7 @@ class TfqPsWeightsFromSymbolOp : public tensorflow::OpKernel {
                                               DoWork);
 
     int largest_single_symbol = 0;
-    for (int i = 0; i < n_single_symbol.size(); i++) {
+    for (size_t i = 0; i < n_single_symbol.size(); i++) {
       largest_single_symbol =
           std::max(n_single_symbol.at(i), largest_single_symbol);
     }
@@ -144,7 +144,7 @@ class TfqPsWeightsFromSymbolOp : public tensorflow::OpKernel {
     auto DoWork2 = [&](int start, int end) {
       for (int i = start; i < end; i++) {
         for (int j = 0; j < n_symbols; j++) {
-          for (int k = 0; k < output_results.at(i).at(j).size(); k++) {
+          for (size_t k = 0; k < output_results.at(i).at(j).size(); k++) {
             output_tensor(i, j, k) = output_results.at(i).at(j).at(k);
           }
           for (int k = output_results.at(i).at(j).size();

--- a/tensorflow_quantum/core/src/program_resolution.cc
+++ b/tensorflow_quantum/core/src/program_resolution.cc
@@ -58,7 +58,7 @@ Status ResolveQubitIds(Program* program,
   std::sort(ids.begin(), ids.end());
 
   absl::flat_hash_map<std::string, int> id_to_index;
-  for (int i = 0; i < ids.size(); i++) {
+  for (size_t i = 0; i < ids.size(); i++) {
     id_to_index[ids[i]] = i;
   }
 
@@ -74,7 +74,7 @@ Status ResolveQubitIds(Program* program,
   }
 
   if (p_sums) {
-    for (int i = 0; i < p_sums->size(); i++) {
+    for (size_t i = 0; i < p_sums->size(); i++) {
       // Replace the PauliSum Qubit ids with the indices.
       for (PauliTerm& term : *(p_sums->at(i)).mutable_terms()) {
         for (PauliQubitPair& pair : *term.mutable_paulis()) {

